lib/compat/wordpress-6.0/edit-site-routes-backwards-compat.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * Site editor's Menu.
+ *
+ * Adds a new wp-admin menu item for the Site editor.
+ *
+ * @since 9.4.0
+ */
+function gutenberg_site_editor_menu() {
+	if ( wp_is_block_theme() ) {
+		add_theme_page(
+			__( 'Editor (beta)', 'gutenberg' ),
+			sprintf(
+			/* translators: %s: "beta" label. */
+				__( 'Editor %s', 'gutenberg' ),
+				'<span class="awaiting-mod">' . __( 'beta', 'gutenberg' ) . '</span>'
+			),
+			'edit_theme_options',
+			'gutenberg-edit-site',
+			'gutenberg_edit_site_page'
+		);
```

Comment:
I'm not 100% sure but this 'hack' could register the page in `$_registered_pages` that is used in `user_can_access_admin_page`.
```suggestion
		add_submenu_page( 'themes.php', null, null, 'edit_theme_options', 'gutenberg-edit-site', 'gutenberg_edit_site_page' );
```

Reply:
I think we'll have to you a similar "hack" to avoid having two menu items.

Reply:
I know the current way is less than ideal. I didn't find a better way of "whitelisting" the routes though. 

> I'm not 100% sure if we should use this 'hack', but it can register the page in $_registered_pages that is used in user_can_access_admin_page, without adding a menu item.

Could you elaborate on this one? Perhaps provide a (pseudo)code snippet exemplifying what you have in mind?

Reply:
My suggestion above does the trick:

>add_submenu_page( 'themes.php', null, null, 'edit_theme_options', 'gutenberg-edit-site', 'gutenberg_edit_site_page' );

that's the hack 😄

Reply:
Ahh, thanks and sorry for missing it!